packages/medusa/src/api-v2/utils/middlewares/pricing/set-pricing-context.ts

```diff
@@ -0,0 +1,88 @@
+import { MedusaPricingContext } from "@medusajs/types"
+import { isPresent, MedusaError } from "@medusajs/utils"
+import { NextFunction } from "express"
+import { MedusaRequest } from "../../../../types/routing"
+import { refetchEntities, refetchEntity } from "../../refetch-entity"
+
+export function setPricingContext() {
```

Comment:
If this is specific to products, maybe we should move it there? There are a lot of assumptions on the req data made here that it doesn't seem like we can have it as a generic middleware?

Reply:
fair point, will move it there